cmd/provision/legacy.go

```diff
@@ -233,13 +233,13 @@ func (p *provision) verifyInternalProxy(client *http.Client, printf shared.Forma
 		q.Add("encrypt", "true")
 		req.URL.RawQuery = q.Encode()
 	}
-	if err != nil {
+	if err == nil {
 		res, err = client.Do(req)
 		if res != nil {
 			defer res.Body.Close()
 		}
 	}
-	if err != nil {
+	if res.StatusCode > 299 || err != nil {
```

Comment:
Do we know for certain that res will not be nil?

Reply:
Good point. Will add the null check.